tests/phpunit/integration/Core/Storage/Has_Connected_AdminsTest.php

```diff
@@ -0,0 +1,72 @@
+<?php
+/**
+ * Has_Connected_AdminsTest
+ *
+ * @package   Google
+ * @copyright 2020 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Tests\Core\Storage;
+
+use Google\Site_Kit\Context;
+use Google\Site_Kit\Core\Storage\Options;
+use Google\Site_Kit\Core\Storage\User_Options;
+use Google\Site_Kit\Core\Storage\Has_Connected_Admins;
+use Google\Site_Kit\Tests\TestCase;
+use WP_Error;
+
+/**
+ * @group Storage
+ */
+class Has_Connected_AdminsTest extends TestCase {
+
+	/**
+	 * @var Context
+	 */
+	protected $context;
+
+	/**
+	 * @var Options
+	 */
+	protected $options;
+
+	/**
+	 * @var User_Options
+	 */
+	protected $user_options;
+
+	public function setUp() {
+		parent::setUp();
+		$this->context      = new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE );
+		$this->options      = new Options( $this->context );
+		$this->user_options = new User_Options( $this->context );
+	}
+
+	public function test_get_type() {
+		$setting = new FakeHas_Connected_Admins( $this->options, $this->user_options );
+		$this->assertEquals( 'boolean', $setting->get_type() );
+	}
+
+	public function test_get__without_option_value_yet() {
+		$setting = new FakeHas_Connected_Admins( $this->options, $this->user_options );
+
+		$setting->set_query_connected_admins_return_value( array() );
+		delete_option( Has_Connected_Admins::OPTION );
+		$this->assertFalse( $setting->get() );
+
+		$setting->set_query_connected_admins_return_value( array( 1 ) );
+		delete_option( Has_Connected_Admins::OPTION );
+		$this->assertTrue( $setting->get() );
+		$this->assertTrue( get_option( Has_Connected_Admins::OPTION ) );
+	}
+
+	public function test_get__with_option_value() {
+		$setting = new FakeHas_Connected_Admins( $this->options, $this->user_options );
+		update_option( Has_Connected_Admins::OPTION, true );
+
+		$this->assertTrue( $setting->get() );
+	}
+
+}
```

Comment:
@aaemnnosttv what's the point to test a function from WP core? IMO we need to make sure that the Has_Connected_Admins class behaves accordingly when it receives certains results from the `get_users` function. That's why it's enough to just mock `get_users`.

Reply:
@eugene-manuilov the point is not to test WP core functionality (if that were the case we would have a lot of missing tests!) but to ensure that the query is called properly. As it is now, any number of changes could be made to the query or method which would break the intended functionality of this class, but the test would continue to pass as written. Another reason is to avoid testing a different version of the class that we don't use rather than testing the real class when it is possible to do so without much cost. Yes, it will take some small amount of time to create users and query them a few times, but in my opinion the benefit (confidence) far outweighs the cost (a very small amount of time and comparable amount of code). This is also consistent with the approach to how many other classes are tested across the codebase.

Reply:
Ok, got it. Updated.